### PR TITLE
Move dynamic workflow metadata to GetNodeExecutionData response

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Move dynamic workflow metadata to GetNodeExecutionData response.

**Note** this isn't backwards compatible but nothing is even referencing DynamicWorkflowNodeMetadata at the moment.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/831

## Follow-up issue
_NA_

### CODE REVIEW

**Review by EngHabu — APPROVED** [review]

File: protos/flyteidl/admin/node_execution.proto [hunk 1]
```diff
@@ -184,4 +181,7 @@ message NodeExecutionGetDataResponse {
 
     // Optional, full_outputs will only be populated if they are under a configured size threshold. 
     core.LiteralMap full_outputs               = 4;
+
+    // In the case this node yielded a task which launched a dynamic workflow, we return its structure here.
```
Comment (kumare3) [inline protos/flyteidl/admin/node_execution.proto:185]:
```suggestion
    // Optional Workflow closure for a dynamically generated workflow, in the case this node yields a dynamic workflow we return its structure here.
```

Question, for retries this will be updated?

Comment (katrogan) [reply protos/flyteidl/admin/node_execution.proto:185]: done. unfortunately no, this is because the task that yields the workflow will be retried rather than the node itself.

**Comment by katrogan** [comment]
PTAL @kumare3

**Review by kumare3 — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]
